internal/logging/rotate.go

```diff
@@ -69,22 +69,30 @@ func rotateLogsOnDisk() {
 	}
 }
 
-// RotateLogListener rotates logs on a timer.
+var stopTimer chan bool
+
 func StartRotateLogTimer() {
-	timeout := 1 * time.Minute
-	if durationString := os.Getenv(constants.SvcLogRotateTimerDurationEnvVarName); durationString != "" {
+	interval := 1 * time.Minute
+	if durationString := os.Getenv(constants.SvcLogRotateIntervalEnvVarName); durationString != "" {
 		if duration, err := strconv.Atoi(durationString); err == nil {
-			timeout = time.Duration(duration) * time.Second
+			interval = time.Duration(duration) * time.Millisecond
 		}
 	}
 
+	stopTimer = make(chan bool)
 	go func() {
 		rotateLogsOnDisk()
 		for {
 			select {
-			case <-time.After(timeout):
+			case <-time.After(interval):
 				rotateLogsOnDisk()
+			case <-stopTimer:
+				return
```

Comment:
```suggestion
			case <-stopTimer:
				return
			case <-time.After(interval):
				rotateLogsOnDisk()
```

I realize both of these triggering in the same loop is pretty unlikely, but still worth being mindful of.